packages/api/cms-api/src/dam/files/files.service.ts

```diff
@@ -247,6 +250,13 @@ export class FilesService {
         const folderId = input.folderId !== undefined ? input.folderId : entity.folder?.id;
         const folder = folderId ? await this.foldersService.findOneById(folderId) : null;
 
+        if (input.name) {
+            const filenameValidationError = this.fileValidationService.validateFilename(input.name, entity.mimetype);
+            if (filenameValidationError) {
+                throw new CometValidationException(filenameValidationError);
+            }
+        }
```

Comment:
Why not validate on the input with a custom validator?

Reply:
I couldn't find a way because I can only access the input object in the decorator. But neither the mimetype nor the id are in the input object, so I couldn't figure out the mimetype. Having the mimetype in input would be unsave anyway. Since the id isn't in input I can't load the file from the DB

Reply:
You could wrap id and input in an additional args class. But it's okay this way.

Reply:
You mean like this? eee61b027f39a93c91f365d432f414f2b4fd642d